### ISSUE TITLE
0.65 TypeError: AsyncClient.__init__() got an unexpected keyword argument 'app'

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#2443]
Just stared seeing this error today after the release of `httpx` 0.28.0:

```
File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/datasette/app.py", line 1601, in request
    async with httpx.AsyncClient(app=self.app) as client:
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: AsyncClient.__init__() got an unexpected keyword argument 'app'
```

**Comment by simonw** [comment]
I fixed this code in `main` but I never backported the fix to the `0.65` release line. Here's the code at fault:

https://github.com/simonw/datasette/blob/753ddc902881b8d5347675c8544f5e10c977fbda/datasette/app.py#L1571-L1573

In `main` it now looks like this:

https://github.com/simonw/datasette/blob/e85517dab3172b82e385e2a366f97a7cd8787e30/datasette/app.py#L2015-L2020

**Comment by simonw** [comment]
I don't think depending on `httpx>=0.28.0` is strictly necessary here - in fact it could break things for people who depend on Datasette stable but also have some other dependency which isn't yet ready for the breaking changes in `httpx`.

(Also the tests just failed because my Pyodide test currently ins to 0.23.0)

**Comment by simonw** [comment]
Released in 0.65.1.
